src/lib/schema.js

```diff
@@ -0,0 +1,147 @@
+import type { ObjectSchema, SchemaRules } from './compose';
+
+import { all, allWhileOK } from './compose';
+
+import {
+  validateIsObject,
+  validateObjHasKey,
+  validateObjPropHasType,
+  validateObjPropPasses,
+  validateObjOnlyHasKeys,
+  validateIsArray,
+  validateArrayItemsHaveType,
+  validateArrayItemsPass,
+  alwaysErr,
+} from './validators';
+
+import { isErr, err, mapErrors, getErrors } from './result';
+
+import { flatten, isObject } from './helpers';
+
+type ConvertToValidator = String => Validator;
+const convertToValidator: ConvertToValidator = key => {
+  if (key === 'required') return validateObjPropHasType('boolean')('required');
+  if (key === 'type') return validateObjPropHasType('string')('type');
+  return validateObjPropHasType('function')('validator');
```

Comment:
Support for the `validator` key here is implicit and not very obvious – I think it'd be a lot clearer if we added another `if` and then pick a sensible default in case the key doesn't match any of the cases.